esp-wifi/src/wifi/mod.rs

```diff
@@ -340,6 +343,196 @@ impl Default for ClientConfiguration {
     }
 }
 
+#[allow(clippy::type_complexity)]
+static CSI_CB: Mutex<RefCell<Option<fn(crate::binary::include::wifi_csi_info_t)>>> =
+    Mutex::new(RefCell::new(None));
+
+unsafe extern "C" fn promiscuous_csi_rx_cb(
+    _ctx: *mut crate::wifi::c_types::c_void,
```

Comment:
At this point we should allocate the callback and pass it via `ctx`. We don't need a critical section here, especially that we don't know how long the callback takes.

The second part can also be solved by moving the function call out of the critical section, too.